honeybee_radiance/lightsource/dictutil.py

```diff
@@ -0,0 +1,56 @@
+# coding=utf-8
+"""Utilities to convert light source dictionaries to Python objects."""
+from honeybee_radiance.lightsource.sunpath import Sunpath
+from honeybee_radiance.lightsource.ground import Ground
+from honeybee_radiance.lightsource.sky.certainirradiance import CertainIrradiance
+from honeybee_radiance.lightsource.sky.cie import CIE
+from honeybee_radiance.lightsource.sky.climatebased import ClimateBased
+from honeybee_radiance.lightsource.sky.hemisphere import Hemisphere
+from honeybee_radiance.lightsource.sky.skydome import Skydome
+from honeybee_radiance.lightsource.sky.skymatrix import SkyMatrix
+from honeybee_radiance.lightsource.sky.sunmatrix import SunMatrix
+
+
+LIGHT_SOURCE_TYPES = \
+    ('Sunpath', 'Ground', 'CertainIrradiance', 'CIE', 'ClimateBased',
+     'Hemisphere', 'SkyDome', 'SkyMatrix', 'SunMatrix')
```

Comment:
What if we change this to a dictionary and use a quick look up instead of going through all the if / else to find the type. This should also keep the performance the same even if we add so many new types:

```python
LIGHT_SOURCE_TYPES = \
    {'Sunpath': Sunpath, 'Ground': Ground, 'CertainIrradiance': CertainIrradiance, 'CIE': CIE, 'ClimateBased': ClimateBased,
     'Hemisphere': Hemisphere, 'SkyDome': SkyDome, 'SkyMatrix': SkyMatrix, 'SunMatrix': SunMatrix}
```

Then for the function itself the code can be like this:

```python

def dict_to_light_source(light_source_dict, raise_exception=True):
    try:  # get the type key from the dictionary
        light_type = light_source_dict['type']
    except KeyError:
        raise ValueError('Load dictionary lacks required "type" key.')

    try:
        # I think this should work otherwise you can use getattr
        return LIGHT_SOURCE_TYPES[light_ype].from_dict(light_source_dict)
    except KeyError:
        raise ValueError(
            '{} is not a recognized radiance Light Source type'.format(light_type))
```

Reply:
I would love to do that except that the name of the class you made is called `Skydome` while the type key is called `SkyDome`.  Would you want me to make the change to `SkyDome` as part of this PR? Then I can implement what you have here.

Reply:
Actually, I guess I can still get the dictionary to work that way since the key can be different from the value. Do you want me to make that change anyway so that we don't make mistakes like the one in your sample in the future?

Reply:
Yes. I just copy/pasted the keys. You can use the correct class names. Didn't really test the code.

Reply:
Done!